src/otx/core/model/entity/classification.py

```diff
@@ -29,9 +30,52 @@
     from torch import device, nn
 
 
-class OTXMulticlassClsModel(
-    OTXModel[MulticlassClsBatchDataEntity, MulticlassClsBatchPredEntity],
-):
+class ExplainableOTXClsModel(OTXModel[T_OTXBatchDataEntity, T_OTXBatchPredEntity]):
```

Comment:
Isn't it possible to attach the hook to an existing classification model? I was wondering if it is really needed to create an additional object for this? I have a concern that there is a growing list of tasks/objects we have, which will ultimately be getting a bit confusing to users.